scrapy/xlib/tx/client.py

```diff
@@ -975,7 +975,7 @@ class ContentDecoderAgent(object):
     def __init__(self, agent, decoders):
         self._agent = agent
         self._decoders = dict(decoders)
-        self._supported = ','.join([decoder[0] for decoder in decoders])
+        self._supported = ','.join(decoder[0] for decoder in decoders)
```

Comment:
do not modify scrapy.xlib.client code, it is a verbatim copy of twisted sources for the specific case of supporting http1.1 in old twisted version.

We are more close to remove this code than improving it.